repos/system_upgrade/el7toel8/actors/checkfstaboptions/actor.py

```diff
@@ -0,0 +1,76 @@
+from collections import defaultdict
+
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecutionError
+from leapp.models import StorageInfo
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, ChecksPhaseTag
+from leapp import reporting
+
+
+class CheckFstabOptions(Actor):
+    """
+    Check for mount options that are invalid under RHEL 8.
+
+    Several XFS mount options have been removed from the kernel between RHEL 7 and 8.
+    Their occurence will lead to an unbootable machine. If one of them is detected, the upgrade is inhibited.
+
+    The list of removed options is taken from the xfs(5) manual page on RHEL 8.
+    """
+
+    name = 'check_fstab_options'
+    consumes = (StorageInfo,)
+    produces = (Report,)
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+
+    REMOVED_XFS_OPTIONS = [
+        # removed from kernel in 4.0
+        'delaylog',
+        'nodelaylog',
+        'ihashsize',
+        'irixsgid',
+        'osyncisdsync',
+        'osyncisosync',
+        # removed from kernel in 4.19
+        'barrier',
+        'nobarrier'
+    ]
+
+    def process(self):
+        storage_info = next(self.consume(StorageInfo), None)
+        if not storage_info:
+            raise StopActorExecutionError(
+                'Could not check fstab entries', details={'details': 'No StorageInfo facts found.'}
+            )
+
+        found = defaultdict(list)
+        for entry in storage_info.fstab:
+            if entry.fs_vfstype == 'xfs':
+                for option in entry.fs_mntops.split(','):
+                    option_name = option.split('=')[0]
+                    if option_name in self.REMOVED_XFS_OPTIONS:
+                        found[entry.fs_spec].append(option_name)
+
+        if found:
+            reporting.create_report([
+                reporting.Title(
+                    'Removed mount options found in /etc/fstab'
```

Comment:
@drehak the title is kind of encrypted. You are referring just to XFS, but title says it's generic.

Reply:
I am curious whether additional FSs have some changes as well.

Reply:
Yeah, I deliberately decided to keep the actor more generic because other FS types might have removed mount options too - now or later. At this moment it only covers XFS.

Reply:
The report title is too cryptic for a bystander, I'd rename to "Unsupported mount options in /etc/fstab detected"